### PR TITLE
sql: update docs for split partition table

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What is changed, added or deleted? (Required)

update docs for `split table` statement.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/tidb/pull/12213

### CODE REVIEW

**Comment by TomShawn** [comment]
@crazycs520 Please involve a technical review.

**Comment by crazycs520** [comment]
@tiancaiamao PTAL

File: reference/sql/statements/split-region.md [hunk 1]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+为 `p1` 分区的 [0,10000] 预切分 2 个 region：
+
+{{< copyable "sql" >}}
+
+```sql
+split partition table t partition (p1) between (0) and (10000) regions 2;
+```
+
+为 `p2` 分区的 [10000,20000] 预切分 4 个 region：
```
Comment (crazycs520) [inline reference/sql/statements/split-region.md:291]:
```suggestion
为 `p2` 分区的 [10000,20000] 预切分 2 个 region：
```

File: reference/sql/statements/split-region.md [hunk 2]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+| 1986      | t_1401_       | t_1401_r      | 1989      | 5               | 1989, 2018, 2019 | 0          | 123           | 0          | 1                    | 0                |
++-----------+---------------+---------------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
+```
+
+也可以给每个分区的索引切分 region ，如将索引 `idx` 的 [1000,10000] 范围切分成 2 个 region：
+
+{{< copyable "sql" >}}
+
+```sql
+split partition table t index idx between (1000) and (10000) regions 2;
```
Comment (tiancaiamao) [inline reference/sql/statements/split-region.md:265]: Split region for index functions on all the partitions rather than on a specified partition?

**Comment by tiancaiamao** [comment]
LGTM

File: reference/sql/statements/split-region.md [hunk 3]
```diff
@@ -27,6 +27,19 @@ SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
 
 `BY value_list…` 语法将手动指定一系列的点，然后根据这些指定的点切分 Region，适用于数据不均匀分布的场景。
 
+Split 语句的返回结果的示例如下：
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:30]:
```suggestion
`SPLIT` 语句的返回结果示例如下：
```

File: reference/sql/statements/split-region.md [hunk 4]
```diff
@@ -27,6 +27,19 @@ SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
...
+
+```sql
++--------------------+----------------------+
+| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
++--------------------+----------------------+
+| 4                  | 1.0                  |
++--------------------+----------------------+
+```
+
+* `TOTAL_SPLIT_REGION`：表示新增预切分的 region 数量。
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:40]:
```suggestion
* `TOTAL_SPLIT_REGION`：表示新增预切分的 Region 数量。
```

File: reference/sql/statements/split-region.md [hunk 5]
```diff
@@ -27,6 +27,19 @@ SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
...
+* `SCATTER_FINISH_RATIO` ：表示新增预切分 region 中，打散完成的比率。如 `1.0` 表示全部完成。`0.5`表示只有一半的 region 已经打散完成了，剩下的还在打散过程中。
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:41]:
```suggestion
* `SCATTER_FINISH_RATIO`：表示新增预切分 Region 中，打散完成的比率。如 `1.0` 表示全部完成。`0.5`表示只有一半的 Region 已经打散完成，剩下的还在打散过程中。
```

File: reference/sql/statements/split-region.md [hunk 6]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
 region4  [("c", "")                    , maxIndexValue               )
 ```
 
+### Split Partition Table Region
+
+预切分分区表的 REGION 使用上和普通表一样，差别是会为每一个 partition 都做相同的切分。
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:186]: 上面的三级标题里没有体现 `预`，确认一下是“预切分”还是“切分”呢？
```suggestion
预切分分区表的 Region 使用上和普通表一样，差别是会为每一个 partition 都做相同的切分。
```

File: reference/sql/statements/split-region.md [hunk 7]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+均匀切分的语法如下：
+
+```sql
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:190]:
```suggestion
{{< copyable "sql" >}}

```sql
```

File: reference/sql/statements/split-region.md [hunk 8]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+SPLIT [PARTITION] TABLE t [PARTITION] [(partition_name_list...)] [INDEX index_name] BETWEEN (lower_value) AND (upper_value) REGIONS region_num
+```
+
+不均匀切分的语法如下：
+
+```sql
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:196]:
```suggestion
{{< copyable "sql" >}}

```sql
```

File: reference/sql/statements/split-region.md [hunk 9]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+首先创建一个分区表：
+
+{{< copyable "sql" >}}
+
+```sql
+create table t (a int,b int,index idx(a)) partition by hash(a) partitions 2;
+```
+
+此时建完表后会为每个 partition 都单独 split 一个 region，用 `SHOW TABLE REGIONS` 语法查看该表的 REGION 如下：
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:210]:
```suggestion
此时建完表后会为每个 partition 都单独 split 一个 Region，用 `SHOW TABLE REGIONS` 语法查看该表的 Region 如下：
```

File: reference/sql/statements/split-region.md [hunk 10]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+{{< copyable "sql" >}}
+
+```sql
+show table t regions;
+```
+
+```sql
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:218]:
```suggestion
```
```

File: reference/sql/statements/split-region.md [hunk 11]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
++-----------+-----------+---------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
+| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS            | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
++-----------+-----------+---------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
+| 1978      | t_1400_   | t_1401_ | 1979      | 4               | 1979, 1980, 1981 | 0          | 0             | 0          | 1                    | 0                |
+| 6         | t_1401_   |         | 17        | 4               | 17, 18, 21       | 0          | 223           | 0          | 1                    | 0                |
++-----------+-----------+---------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
+```
+
+再用 `SPLIT` 语法为每个 partition 切分 region，示例如下，在 [0,10000] 范围内切分成 4 个 region：
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:227]:
```suggestion
再用 `SPLIT` 语法为每个 partition 切分 Region，示例如下，在 [0,10000] 范围内切分成 4 个 Region：
```

File: reference/sql/statements/split-region.md [hunk 12]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+{{< copyable "sql" >}}
+
+```sql
+split partition table t between (0) and (10000) regions 4;
+```
+
+再用 `SHOW TABLE REGIONS` 语法查看该表的 REGION 如下，会发现该表现在一共有 10 个 region，每个 partition 分别有 5 个 region，其中 4 个 region 是表的行数据，1 个 region 是表的索引数据。
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:235]:
```suggestion
再用 `SHOW TABLE REGIONS` 语法查看该表的 Region。如下会发现该表现在一共有 10 个 Region，每个 partition 分别有 5 个 Region，其中 4 个 Region 是表的行数据，1 个 Region 是表的索引数据。
```

File: reference/sql/statements/split-region.md [hunk 13]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+{{< copyable "sql" >}}
+
+```sql
+show table t regions;
+```
+
+```sql
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:243]:
```suggestion
```
```

File: reference/sql/statements/split-region.md [hunk 14]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+| 1990      | t_1401_r      | t_1401_r_2500 | 1992      | 2               | 1991, 1992, 2020 | 0          | 120           | 0          | 1                    | 0                |
+| 1994      | t_1401_r_2500 | t_1401_r_5000 | 1997      | 5               | 1996, 1997, 2021 | 0          | 129           | 0          | 1                    | 0                |
+| 2002      | t_1401_r_5000 | t_1401_r_7500 | 2003      | 4               | 2003, 2023, 2022 | 0          | 141           | 0          | 1                    | 0                |
+| 6         | t_1401_r_7500 |               | 17        | 4               | 17, 18, 21       | 0          | 601           | 0          | 1                    | 0                |
+| 1986      | t_1401_       | t_1401_r      | 1989      | 5               | 1989, 2018, 2019 | 0          | 123           | 0          | 1                    | 0                |
++-----------+---------------+---------------+-----------+-----------------+------------------+------------+---------------+------------+----------------------+------------------+
+```
+
+也可以给每个分区的索引切分 region ，如将索引 `idx` 的 [1000,10000] 范围切分成 2 个 region：
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:260]:
```suggestion
也可以给每个分区的索引切分 Region ，如将索引 `idx` 的 [1000,10000] 范围切分成 2 个 Region：
```

File: reference/sql/statements/split-region.md [hunk 15]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+{{< copyable "sql" >}}
+
+```sql
+split partition table t index idx between (1000) and (10000) regions 2;
+```
+
+#### Split 单个分区 的 region
+
+可以单独指定要 切分的 partition，示例如下：
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:270]:
```suggestion
可以单独指定要切分的 partition，示例如下：
```

File: reference/sql/statements/split-region.md [hunk 16]
```diff
@@ -168,6 +181,149 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+也可以给每个分区的索引切分 region ，如将索引 `idx` 的 [1000,10000] 范围切分成 2 个 region：
+
+{{< copyable "sql" >}}
+
+```sql
+split partition table t index idx between (1000) and (10000) regions 2;
+```
+
+#### Split 单个分区 的 region
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:268]:
```suggestion
#### Split 单个分区的 Region
```

File: reference/sql/statements/split-region.md (hunk 13 above)
Comment (crazycs520) [reply reference/sql/statements/split-region.md:247]: Why remove this line?

File: reference/sql/statements/split-region.md (hunk 2 above)
Comment (crazycs520) [reply reference/sql/statements/split-region.md:269]: yes,

File: reference/sql/statements/split-region.md (hunk 10 above)
Comment (crazycs520) [reply reference/sql/statements/split-region.md:222]: why remove this?

File: reference/sql/statements/split-region.md (hunk 6 above)
Comment (crazycs520) [reply reference/sql/statements/split-region.md]: 这里都行，从 split 字面翻译可能 ”分区“ ，不要"预"先 更好些。

**Pull request merged**